src/upload-emoji.js

```diff
@@ -1,36 +1,22 @@
-import _ from 'lodash';
-import superagent from 'superagent';
-import SuperagentThrottle from 'superagent-throttle';
-import uuid from 'uuid';
-
+import axios from 'axios';
 import getSlackApiData from './get-slack-api-data';
-
-const NO_OP = function () {};
-
-const superagentThrottle = new SuperagentThrottle({
-  active: true,
-  concurrent: 5,
-  rate: Infinity
-});
-
-export default function uploadEmoji (file, callback = NO_OP) {
-  const { apiToken, versionUid } = getSlackApiData();
+export default function uploadEmoji(file) {
+  const { token, versionUid } = getSlackApiData();
   const timestamp = Date.now() / 1000;  
   const version = versionUid ? versionUid.substring(0, 8) : 'noversion';
-  const id = uuid.v4();
+  
   const name = file.name.split('.')[0];
-  const imageUploadRequest = superagent.post('/api/emoji.add')
-    .withCredentials()
-    .query(`_x_id=${version}-${timestamp}`)
-    .field('name', name)
-    .field('mode', 'data')
-    .field('token', apiToken)
-    .attach('image', file)
-    .use(superagentThrottle.plugin())
-    .end((error, response) => {
-      const uploadError = error || _.get(response.body, 'error');
-      callback(uploadError, response);
-    });
 
-  return id;
-}
\ No newline at end of file
+  const formData = new FormData();
+  formData.append('_x_id', `${version}-${timestamp}`)
+  formData.append('name', name);
+  formData.append('token', token);
+  formData.append('mode', 'data');
+  formData.append('image', file);
```

Comment:
I believe this is required for an image upload with axios. I tried just passing these key: values over in an object but didn't have any luck. This part definitely isn't as graceful as superagent was.